Language/Classes/Constructors/Factories/inference_A01_t01.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The return type of a factory whose signature is of the form
+/// `factory M` or the form `factory M.id` is `M` if `M` is not a generic type;
+/// otherwise the return type `M<T1,..., Tn>` where `T1,...,Tn` are the type
+/// parameters of the enclosing class.
+///
+/// @description Checks that return type of the factory constructor can be
+/// inferred.
```

Comment:
I don't think the issue was that the return type was now and then not inferred correctly, the issue was that the redirectee `D._` is subject to type inference, but in some situations (such as when the analyzer created a summary of `inference_lib.dart`) no type inference was performed. Instead, the analyzer used instantiation to bound. This caused the redirection to be treated as `D<Object>._` rather than `D<T>._`, and that's an error because the return type of `D<Object>._` is `D<Object>`, which is not a subtype of the return type of the redirecting factory which is `C0<T>`.

So we're actually checking that the redirectee type inference is performed (as opposed to being replaced by instantiation to bound).